trecs/models/homogenizer.py

```diff
@@ -0,0 +1,221 @@
+
+        >>> hm = Homogenizer()
+        >>> hm.users_hat.shape
+        (100, 10)   # <-- 100 users (default), 10 attributes (default)
+        >>> hm.items.shape
+        (10, 1250) # <-- 10 attributes (default), 1250 items (default)
+
+        This class can be customized either by defining the number of users/items/attributes
+        in the system as well as by the specified model class.
+
+        >>> hm = Homogenizer(num_users=1200, num_items=5000, model_class=ImplicitMF)
+        >>> hm.users_hat.shape
+        (1200, 10) # <-- 1200 users, 10 attributes
+
+        >>> hm = Homo(num_users=1200, num_items=5000, num_attributes=2000)
+        >>> hm.users_hat.shape
+        (1200, 2000) # <-- 1200 users, 2000 attributes
+
+        Or by generating representations for items and/or users. In the example
+        below, items are uniformly distributed. We indirectly define 100
+        attributes by defining the following `item_representation`:
+        >>> number_of_users = 5
+        >>> number_of_attributes = 10
+        >>> number_of_items = 15
+        >>> users = np.random.randint(4, size=(number_of_users, number_of_attributes))
+        >>> items = Generator().binomial(n=1, p=.3,size=(number_of_attributes, number_of_items))
+        >>> hm = Homogenizer(item_representation=items, user_representation=users)
+        >>> hm.items.shape
+        (10, 15)
+        >>> hm.users_hat.shape
+        (5, 10)
+
+        Note that all arguments passed in at initialization must be consistent -
+        otherwise, an error is thrown. For example, one cannot pass in
+        `num_users=200` but have `user_representation.shape` be `(300, 100)`.
+        Likewise, one cannot pass in `num_items=1000` but have
+        `item_representation.shape` be `(100, 500)`.
+    """
+
+    def __init__(  # pylint: disable-all
+        self,
+        num_users=None,
+        num_items=None,
+        num_attributes=10,
+        user_representation=None,
+        item_representation=None,
+        actual_user_representation=None,
+        actual_item_representation=None,
+        probabilistic_recommendations=False,
+        seed=None,
+        verbose=False,
+        num_items_per_iter=10,
+        homogenization_increment=0.1,
+        model_class=ContentFiltering,
+        **kwargs
+    ):
+
+        self.__class__ = type(
+            self.__class__.__name__, (model_class, object), dict(self.__class__.__dict__)
+        )
+
+        # measurements = [MSEMeasurement()]
+
+        super(self.__class__, self).__init__(
+            num_users,
+            num_items,
+            num_attributes,
+            user_representation,
+            item_representation,
+            actual_user_representation,
+            actual_item_representation,
+            probabilistic_recommendations,
+            seed,
+            verbose,
+            num_items_per_iter,
+            **kwargs
+        )
+
+        assert 0 < homogenization_increment < 1
```

Comment:
can we turn this into an `if` statement that explicitly raises an exception if the condition is not satisfied?